blocks/utils.py

```diff
@@ -405,8 +405,56 @@ def ipdb_breakpoint(x):
     import ipdb
     ipdb.set_trace()
 
+pickling_error = """
 
-def secure_dill_dump(object_, path):
+Blocks relies on the ability to pickle the entire main loop, which includes \
+all bricks, data streams, extensions, etc. All of these must be serializable \
+using Pythons pickling library. This means certain things such as nested \
+functions, lambda expressions, generators, etc. should be avoided. Please \
+see the documentation for more detail."""
+
+instancemethod_error = """
+
+Python is unable to serialize references to class and instance methods. This \
+means that you cannot store reference to objects like `str.lower` (a class \
+method) or `Linear().allocate` (an instance method) in an attribute. If you \
+reall need to do this, it is often possible to write a wrapper function such \
+as `def lower(s): return s.lower()` and store that instead."""
+
+lambda_error = """
+
+Python is unable to serialize lambda functions. Make sure that your code does \
+not rely on them and use normal function definitions instead."""
+
+nested_function_error = """
+
+Python is unable to serialize nested functions. Make sure that all your \
+functions are available in the global namespace."""
+
+
+def pickle_dump(*args, **kwargs):
```

Comment:
I think we should move these functions to something like `blocks/serialization.py`, similarly to `blocks/dump.py`. They are too fundamental to be part of a general util stuff.